### PR TITLE
Fix inconsistency when hashing two tables in `cudf::detail::contains`

### PR DESCRIPTION
**Pull request by ttnghia**
When hashing elements in a column, the nullable information needs to be taken into account. This could produce different results for the hash values of the same elements if the nullable condition is changed. As such, whenever we need to compute hashing for more than one column/table in the same operation, we need to take into account the nullable of all columns/tables, not just the one that is being hashed.

This PR fixes a bug when hashing two tables using different nullable values in `cudf::detail::contains`, which led to incorrect results when there is only one nullable table in the input.

### CODE REVIEW

**Review by bdice — APPROVED** [review]
This fix looks good - it makes sense and I wish I'd caught this in my initial review. :) @ttnghia Would you be able to write a test that catches this? (I see it was caught by Spark integration tests but I'd like a test in libcudf.)

**Comment by ttnghia** [comment]
> This fix looks good - it makes sense and I wish I'd caught this in my initial review. :) @ttnghia Would you be able to write a test that catches this? (I see it was caught by Spark integration tests but I'd like a test in libcudf.)

Yeah the bug seems caused by a not very intuitive reason.
I have added a test for this particular case. It fails with the previous implementation and only passed with this fix.

**Comment by pxLi** [comment]
@gpucibot merge

**Comment by davidwendt** [comment]
@pxLI Please don't merge C++ PRs unless they have at least 2 approvals.

**Comment by pxLi** [comment]
> @pxLi Please don't merge C++ PRs unless they have at least 2 approvals.

sry my bad, I did not realize this rule.  I will take it in mind in the future~

**Pull request merged**
